google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/compute/GoogleComputeEngineServiceAdapter.java

```diff
@@ -286,15 +280,30 @@ public MachineTypeInZone apply(MachineType arg0) {
               .addAll(api.getImageApiForProject(userProject.get()).list().concat())
               .addAll(api.getImageApiForProject(DEBIAN_PROJECT).list().concat())
               .addAll(api.getImageApiForProject(CENTOS_PROJECT).list().concat())
+              .addAll(api.getImageApiForProject(UBUNTU_PROJECT).list().concat())
+              .addAll(api.getImageApiForProject(GOOGLE_CONTAINERS_PROJECT).list().concat())
+              .addAll(api.getImageApiForProject(RHEL_PROJECT).list().concat())
+              .addAll(api.getImageApiForProject(CORE_OS_PROJECT).list().concat())
+              .addAll(api.getImageApiForProject(OPENSUSE_PROJECT).list().concat())
+              .addAll(api.getImageApiForProject(SUSE_PROJECT).list().concat())
+              .addAll(api.getImageApiForProject(WINDOWS_PROJECT).list().concat())
               .build();
    }
 
    @Override
    public Image getImage(String id) {
-      return Objects.firstNonNull(api.getImageApiForProject(userProject.get()).get(id),
-                                  Objects.firstNonNull(api.getImageApiForProject(DEBIAN_PROJECT).get(id),
-                                          api.getImageApiForProject(CENTOS_PROJECT).get(id)));
-
+      Image image;
+      if ((image = api.getImageApiForProject(userProject.get()).get(id)) != null) return image;
+      if ((image = api.getImageApiForProject(DEBIAN_PROJECT).get(id)) != null) return image;
+      if ((image = api.getImageApiForProject(CENTOS_PROJECT).get(id)) != null) return image;
+      if ((image = api.getImageApiForProject(UBUNTU_PROJECT).get(id)) != null) return image;
+      if ((image = api.getImageApiForProject(GOOGLE_CONTAINERS_PROJECT).get(id)) != null) return image;
+      if ((image = api.getImageApiForProject(RHEL_PROJECT).get(id)) != null) return image;
+      if ((image = api.getImageApiForProject(CORE_OS_PROJECT).get(id)) != null) return image;
+      if ((image = api.getImageApiForProject(OPENSUSE_PROJECT).get(id)) != null) return image;
+      if ((image = api.getImageApiForProject(SUSE_PROJECT).get(id)) != null) return image;
+      if ((image = api.getImageApiForProject(WINDOWS_PROJECT).get(id)) != null) return image;
```

Comment:
I think it's better to use [ObjectUtils.firstNonNull](http://commons.apache.org/proper/commons-lang/javadocs/api-3.1/org/apache/commons/lang3/ObjectUtils.html#firstNonNull%28T...%29) if jclouds-labs-google depends on commons-lang, which is probable. If it does not, then this is fine.